### PR TITLE
airbrake-ruby: deprecate `notifier_name` argument

### LINKED ISSUE #165: Get rid of third argument in public API methods
The third argument is `notifier_name`. I never really liked the way it's designed, but it's was simple in terms of implementation. Reminder, this is the old API:

```rb
Airbrake.notify('oops', params, :project_a)
# ...
Airbrake.any_function_call(any_args, :project_a)
```

Instead, I'd like to decouple `:project_a` from method signatures. Proposed way:

```rb
Airbrake[:project_a].notify('oops', params)
```

The old way of calling Airbrake must also be supported: `Airbrake.notify`.

### PR DESCRIPTION
**Pull request by kyrylo**
Partially resolves #165 (Get rid of third argument in public API
methods)

### CODE REVIEW

File: lib/airbrake-ruby.rb [hunk 1]
```diff
@@ -139,7 +151,16 @@ def configure(notifier = :default)
...
+    def notify(exception, params = {}, notifier = (no_arg = true && :default))
+      unless no_arg
+        warn(
+          "#{LOG_LABEL} `Airbrake.notify` method signature was changed. " \
+          "Passing `notifier_name` is deprecated and will be removed in the " \
+          "next MAJOR release.\n" \
+          "Use `Airbrake[:#{notifier}]` to access the :#{notifier} notifier " \
+          "and call same methods directly on it."
+        )
+      end
```
Comment (shifi) [inline lib/airbrake-ruby.rb]: Move this to method and re-use it?

Comment (kyrylo) [inline lib/airbrake-ruby.rb]: 👍

**Comment by kyrylo** [comment]
PTAL

**Comment by grosser** [comment]
would it be simpler to make it part of the options ?
`Airbrake.notify(foo, notifier: :bar)`
(never used this feature)

**Comment by kyrylo** [comment]
@grosser IMO that would be an overkill because we don't have any other arguments to pass.

**Comment by grosser** [comment]
I'm fine with either ... don't really care ... only thing that I could use is if `Airbrake.configuration` started working again.

**Comment by kyrylo** [comment]
@grosser please file a new issue about this.

**Pull request merged**